app/controllers/answers_controller.rb

```diff
@@ -39,6 +39,7 @@ def destroy
 
   def accept
     return unauthorized_access unless @question.user == current_user
+    return invalid_request('Question has an already accepted answer') if @question.answers.any?(&:accepted)
```

Comment:
A `before_action` makes a whole lot of sense. My thoughts however is that having a `before_action` for a **one line** method that is used just by one action is something of an _overkill_